hook_test.go

```diff
@@ -1,3 +1,5 @@
+// +build !enable_binary_log
```

Comment:
hooks does work. these tests compare the output with a json string. So duplicated few of the testcases into binary_tests.go and there i convert binary to json and compare with the output.

other thought i had was to clone each _test.go into a _binary_test.go and do the exact same tests.. but then i decided it will be too many _test.go files - so i went with one binary_tests.go with all binary related tests (hooks, ctxt..)

Reply:
What about wrapping the wanted output into a noop method in json mode that convert the json into cbor in binary mode?

Reply:
Done.